### PR TITLE
Use proper perl "signatures" in backend

### PR DESCRIPTION
**Pull request by okurz**
Using a command like

```
sed -i "/^use strict;/ {N;s/use strict;/use Mojo::Base -strict, -signatures;/}" $(git ls-files)
for i in $(git grep -l 'use \(strict\|warnings\)'); do grep -q 'use Mojo::Base -strict' $i && sed -i '/use \(strict\|warnings\)/d' $i; done
tools/tidy --only-changed
```

and some manual fixes.

### CODE REVIEW

File: doc/backend_vars.asciidoc [hunk 1]
```diff
@@ -289,3 +289,15 @@ AMT_PASSWORD;string;;Password for admin AMT user on target host
...
 |====================
+
+.SPVM backend
+[grid="rows",format="csv"]
+[options="header",cols="^m,^m,^m,v",separator=";"]
+|====================
+Variable;Values allowed;Default value;Explanation
+WORKER_HOSTNAME;string;undef;Worker host name
+NOVALINK_HOSTNAME;string;undef;Novalink target host to connect to
+NOVALINK_USERNAME;string;undef;Username to authenticate on Novalink host
```
Comment (baierjan) [inline doc/backend_vars.asciidoc:300]: Isn't there a default 'root' according to https://github.com/os-autoinst/os-autoinst/pull/1790/files#diff-b3e78e478b3e4f99f0291d2d0f49ea3d73babcf5c712d50f9d6ea770af388bf3R32?

```suggestion
NOVALINK_USERNAME;string;root;Username to authenticate on Novalink host
```

**Comment by kalikiana** [comment]
I see a lot of seemingly unrelated commits and no particular comments or mention in the description - is any of this supposed to be part of other PR's?

**Review by Martchus — CHANGES_REQUESTED** [review]
Besides the problem mentioned in the inline comment it looks good *as far as I can tell*. I've checked many of the invocations but not all.

**Comment by okurz** [comment]
> I see a lot of seemingly unrelated commits and no particular comments or mention in the description - is any of this supposed to be part of other PR's?

I can change it depending on preferences. All the changes I did were done as part of the same work so I did not arbitrarily mash in unrelated changes by accident :) Some people prefer less pull requests, some prefer more separate ones. I now have separated out PRs 
* https://github.com/os-autoinst/os-autoinst/pull/1843
* https://github.com/os-autoinst/os-autoinst/pull/1844
* https://github.com/os-autoinst/os-autoinst/pull/1845

I suggest to start with PRs however they are and if you feel that a PR becomes too big please just ask to split it

File: doc/backend_vars.asciidoc (hunk 1 above)
Comment (okurz) [reply doc/backend_vars.asciidoc:300]: fixed as part of https://github.com/os-autoinst/os-autoinst/pull/1845

File: backend/pvm.pm [hunk 2]
```diff
@@ -80,10 +77,7 @@ sub do_extract_assets {
     qx/sudo rescan-scsi-bus.sh -r/;
 }
 
-sub pvmctl {
-    my $self = shift @_;
-    my $type = shift @_;
-    my $action = shift @_;
+sub pvmctl ($self, $type, $action) {
```
[the remark this answers is not in the document]
Comment (okurz) [reply backend/pvm.pm]: I realized that I only covered backend::spvm with testing, not backend::pvm. Trying to develop a test it feels more and more like backend::pvm is unused. A `select count(*) from job_settings where key='BACKEND' and value = 'pvm';` reveals no hits for both o3 and osd and the git history also looks like everyone for PowerVM is using either pvm_hmc or spvm. I think the best approach would be to declare pvm itself deprecated and fade it out of use. For now I just applied the fix for `do_extract_assets`

[the remark this answers is not in the document]
Comment (okurz) [reply backend/pvm.pm]: created https://progress.opensuse.org/issues/102146 for the deprecation

[the remark this answers is not in the document]
Comment (okurz) [reply backend/pvm.pm]: I added a trivial instantiation test of backend::pvm and in another commit declared the backend as deprecated. I also applied the fix as previously described. I would not put more effort into backend::pvm

[the remark this answers is not in the document]
Comment (okurz) [reply backend/pvm.pm]: If you prefer the separate PR for the deprecation first we can go with https://github.com/os-autoinst/os-autoinst/pull/1848

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**